engine/scheduler.go

```diff
@@ -81,7 +88,18 @@ func (sas sortableAgentStates) Len() int      { return len(sas) }
 func (sas sortableAgentStates) Swap(i, j int) { sas[i], sas[j] = sas[j], sas[i] }
 
 func (sas sortableAgentStates) Less(i, j int) bool {
-	niUnits := len(sas[i].Units)
-	njUnits := len(sas[j].Units)
-	return niUnits < njUnits || (niUnits == njUnits && sas[i].MState.ID < sas[j].MState.ID)
+	iWeight, _ := calculateAgentStatsWeight(sas[i])
```

Comment:
ping @zeisss this is what I mentioned in our previous meeting. We could easily adapt the scheduler to our needs. This implementation was just an experiment :)